### ISSUE TITLE
About training with 8 gpu

### ISSUE THREAD

**Issue by memoiry** [chenxin-dlut/TransT#43]
Hi,

I found that the training time of the final 500 epochs does not change much whether it is 8 cards or 2 cards. In theory, shouldn’t it be accelerated by 4 times with 8 cards?

Thanks!

**Comment by chenxin-dlut** [comment]
you can use ddp training instead of dp training to accelarate

**Comment by memoiry** [comment]
Hi, @chenxin-dlut , thanks for your reply.

Another question is that have you tried to use 8gpu training? Should I change the learning rate If I use batch size: 16 * 8 = 128, could you give some suggestion? thanks!

Best.

**Comment by chenxin-dlut** [comment]
I use the same learning rate for 8gpu, training 500 epochs with 80000 samples each epoch, with batch size 128, and the learning rate drops at 400th epoch.

**Comment by memoiry** [comment]
Got it, @chenxin-dlut . I tried to modify the code to support ddp. According to your 500 epochs with 80000 samples each epoch, with batch size 128, it takes about 44 hours to complete the training. Do you need a pull request? I still have several questions?

1. For model evaluation, I found that there is no code evaluated during the training process in the code. Then how to evaluate the quality of model checkpoint? Do I just need to look at iou metric? Is this indicator probably positively correlated with the final evaluation indicator?
2. If train on got-10k only, what is the final convergent iou?

Thanks!

**Comment by chenxin-dlut** [comment]
Hello, Thank you for your work, but there will be a large update after VOT2021 competition, so we don't need a pull request now.
For your questions:
1. There is no evaluation during training because the loss of the network does not match the performance on benchmarks in tracking field, so I think the evaluation in training is not meaningful. In order to speed up the training, I remove the evaluation in training. For the checkpoint selection, you can refer IoU, and I suggest taking the last few epochs to test and decide which one to choose.
2. About 90.5

**Comment by memoiry** [comment]
Got it, thanks!

**Comment by Tchuanm** [comment]
@memoiry Hi, did you modify the code to support DDP successfully?  I meet some trouble with DistributedSampler.

**Comment by Tchuanm** [comment]
@chenxin-dlut. Thanks for your nice work.  If I train on the 4 training datasets, what is the final convergent iou?

**Comment by iminfine** [comment]
> Got it, @chenxin-dlut . I tried to modify the code to support ddp. According to your 500 epochs with 80000 samples each epoch, with batch size 128, it takes about 44 hours to complete the training (8GPU). Do you need a pull request?
> 
> BTW, I still have several questions.
> 
> 1. For model evaluation, I found that there is no evaluation during the training process, then how to evaluate the quality of model checkpoint? Do I just need to look at iou metric? Is this indicator probably positively correlated with the final evaluation indicator?
> 2. If train on got-10k only, what is the final convergent iou?
> 
> Thanks!

Could you please share the modified code for ddp?
It is too slow for 8 gpus!
Thanks.
